File: DAL/Entities/UploadedCSV.cs
using System.ComponentModel.DataAnnotations.Schema;
using DAL.Entities.Enums;

namespace DAL.Entities;

public class UploadedCSV: BaseEntity
{
    public Guid ClientId { get; set; }

    [ForeignKey(nameof(ClientId))]
    public Client? Client { get; set; }

    public string FileName { get; set; } = string.Empty;
    public string FilePath { get; set; } = string.Empty;
    public DateTime UploadDate { get; set; } = DateTime.Now;
    public CsvStatus Status { get; set; } = CsvStatus.Pending;
}

File: RecifyWeb/src/app/models/uploaded-csv.ts
export class UploadedCsv {
    id: string;
    clientId: string;
    fileName: string;
    filePath: string;
    uploadDate: Date;
    status: number;
    recommenderToUploadedCsvIds: number[];

    constructor(
        id: string,
        clientId: string,
        fileName: string,
        filePath: string,
        uploadDate: Date,
        status: number,
        recommenderToUploadedCsvIds: number[] = []
    ) {
        this.id = id;
        this.clientId = clientId;
        this.fileName = fileName;
        this.filePath = filePath;
        this.uploadDate = uploadDate;
        this.status = status;
        this.recommenderToUploadedCsvIds = recommenderToUploadedCsvIds;
    }
}